src/main/java/de/tum/in/www1/artemis/service/ModelingExerciseService.java

```diff
@@ -39,9 +37,11 @@ public ModelingExerciseService(ParticipationService participationService,
     @Transactional(readOnly = true)
     public ModelingExercise findOne(Long id) {
         log.debug("Request to get Modeling Exercise : {}", id);
-        return modelingExerciseRepository.findById(id).get();
+        return modelingExerciseRepository.findById(id)
+            .orElseThrow(() -> new EntityNotFoundException("Exercise with id: \"" + id + "\" does not exist"));
```

Comment:
This is a good change. However are you sure that the EntityNotFoundException is handled properly in all cases where this method is invoked? Please double check

Reply:
Yep i already did that.